src/build/env.py

```diff
@@ -245,19 +274,19 @@ def install(self, requirements):  # type: (Iterable[str]) -> None
         if not requirements:
             return
 
-        if sys.version_info[0] == 2:
-            subprocess.check_call([self.executable, '-m', 'ensurepip'], cwd=self.path)
-
         with tempfile.NamedTemporaryFile('w+', prefix='build-reqs-', suffix='.txt', delete=False) as req_file:
             req_file.write(os.linesep.join(requirements))
             req_file.close()
             cmd = [
-                self.executable,
+                cast(str, self._pip_executable),
```

Comment:
Please raise an error if `self._pip_executable` is `None` instead. That approach is better at surviving refactors or other changes.

Reply:
The way the code it's written that use case cannot happen... feels redundant to have a branch we don't cover. IMHO the way to ensure we handle various use cases is by tests and coverage, rather than redundant defensive checks.

Reply:
If for anything else, `mypyc` works fine right now, using `typing.cast` will break that. I generally try to keep of `typing.cast` to the absolute minimum.

Also remember that people can subclass this, so I don't think this check would be *that* defensive.

Reply:
>If for anything else, mypyc works fine right now, using typing.cast will break that.

How? It passes for me as is now here.

If they subclass it and break the contract of using the class it's on them. Also if this is None it will fail with an error that's self descriptive, not sure what's the value in raising a custom error.

Reply:
Alright. I still have a slight preference on having the check but go ahead.

Reply:
To be fair I want to move the isolation creation to the ``__init__`` rather than ``__enter__``, and keep the ``__enter__`` almost empty. That way the executable and pip executable are guaranteed to be not None. This class is not useful without the ``__enter__`` so maybe worth doing all the setup within init. But that's orthogonal to this change, so was thinking to do it in a follow-up PR?

Reply:
That seems fine.

Reply:
Sorry if I'm late to the game here, but a low cost option I've taken in the past for stuff like this would be just `assert self._pop is not None`.